---
title: "README"
author: "Gregory J. Botwin"
date: "2/21/2018"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=FALSE, warning=FALSE, message=FALSE)
library(knitr)
library(tidyverse)
```

## Project Goals
I want to study phenotypes that are genetically similar to IBD using the UK Biobank Phentoype Data produced by the Neale Lab http://www.nealelab.is/blog/2017/7/19/rapid-gwas-of-thousands-of-phenotypes-for-337000-samples-in-the-uk-biobank 

## Tasks
### Get list of all IBD associated SNPs
On 2/21/18 I queried the NHGRI-EBI Catalog of published genome-wide association studies for:  
- Inflammatory Bowel Disease  
- p value <= 5x10^-8  
File saved in "data/gwas-association-downloaded_2018-02-21-inflammatory bowel disease.tsv"

A description of the terms can be found http://www.ebi.ac.uk/gwas/docs/methods 

```{r}
ibd_gwas_catalog <- read_tsv("data/gwas-association-downloaded_2018-02-21-inflammatory bowel disease.tsv") 
colnames(ibd_gwas_catalog) <- make.names(colnames(ibd_gwas_catalog))
kable(ibd_gwas_catalog %>%
  group_by(DISEASE.TRAIT) %>%
  summarise(n_snps = length(unique(STRONGEST.SNP.RISK.ALLELE))) %>%
  arrange(desc(n_snps)), caption = "Number of Unique SNPs with P <= 5x10^-8 per IBD Trait")
```

## Determine Genetic Similarity 
"We evaluated genetic similarity between diseases by the number of shared mutations through the exact binominal test. Mutations were assigned to a disease if they associated with either the disease itself or its descendant phenotypes according to the hierarchical relationships of diseases as defined in HPO. For two diseases, A and B are associated with m mutations and n mutations, respectively. The chance of one mutation related with disease A in all mutations existed in the database is m/N, where N is the total number of mutations in the HGMD. Thus, the expected number of mutations shared by both A and B is m Ã— n/N. We excluded pairs of diseases where one disease was a descendant of the other." DOI: 10.1002/humu.23358 